src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -129,6 +129,15 @@ def _offer(self, payload):
             assert set(arguments.keys()) == {'price', 'energy', 'transaction_id'}
             arguments['seller'] = self.device.name
             arguments['seller_origin'] = self.device.name
+            if arguments['energy'] > self.state.energy_to_sell_dict[self.market.time_slot]:
```

Comment:
I think it would be great to also add this check in the `_offer_impl` method too, as a second layer of protection. Reason for that, is in case a new offer is posted in the tick_event, before the incoming message is handled.